cmd/commands/cluster.go

```diff
@@ -0,0 +1,258 @@
+
+func NewClusterCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Hidden:            true, // until app-proxy is working correctly
+		Use:               "cluster",
+		Short:             "Manage clusters of Codefresh runtimes",
+		PersistentPreRunE: cfConfig.RequireAuthentication,
+		Args:              cobra.NoArgs, // Workaround for subcommand usage errors. See: https://github.com/spf13/cobra/issues/706
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.HelpFunc()(cmd, args)
+			exit(1)
+		},
+	}
+
+	cmd.AddCommand(NewClusterAddCommand())
+	cmd.AddCommand(NewClusterListCommand())
+
+	return cmd
+}
+
+func NewClusterAddCommand() *cobra.Command {
+	var (
+		opts ClusterAddOptions
+		err  error
+	)
+
+	cmd := &cobra.Command{
+		Use:     "add RUNTIME_NAME --context CONTEXT_NAME",
+		Short:   "Add a cluster to a given runtime",
+		Args:    cobra.MaximumNArgs(2),
+		Example: util.Doc(`<BIN> cluster add my-runtime --context my-context`),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+
+			ctx := cmd.Context()
+
+			opts.runtimeName, err = ensureRuntimeName(ctx, args)
+			if err != nil {
+				return err
+			}
+
+			opts.kubeContext, err = ensureKubeContextName(cmd.Flag("context"))
+			return err
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runClusterAdd(cmd.Context(), &opts)
+		},
+	}
+
+	cmd.Flags().BoolVar(&opts.dryRun, "dry-run", false, "")
+	opts.kubeFactory = kube.AddFlags(cmd.Flags())
+	die(err)
+
+	return cmd
+}
+
+func runClusterAdd(ctx context.Context, opts *ClusterAddOptions) error {
+	runtime, err := cfConfig.NewClient().V2().Runtime().Get(ctx, opts.runtimeName)
+	if err != nil {
+		return err
+	}
+
+	if runtime.RuntimeVersion == nil {
+		return fmt.Errorf("runtme \"%s\" has no version", opts.runtimeName)
+	}
+
+	version := semver.MustParse(*runtime.RuntimeVersion)
+	if version.LessThan(minimalSupportedVersion) {
+		return fmt.Errorf("runtme \"%s\" does not support this command. Minimal required version is %s", opts.runtimeName, minimalSupportedVersion)
+	}
+
+	if runtime.IngressHost == nil {
+		return fmt.Errorf("runtime \"%s\" is missing an ingress URL", opts.runtimeName)
+	}
+
+	ingressUrl := *runtime.IngressHost
+	server, err := util.KubeServerByContextName(opts.kubeContext)
+	if err != nil {
+		return fmt.Errorf("failed getting server for context \"%s\": %w", opts.kubeContext, err)
+	}
+
+	csdpToken := cfConfig.GetCurrentContext().Token
+	k := createAddClusterKustomization(ingressUrl, opts.runtimeName, server, csdpToken, runtime.RuntimeVersion)
+
+	manifests, err := kustutil.BuildKustomization(k)
+	if err != nil {
+		return fmt.Errorf("failed building kustomization:%w", err)
+	}
+
+	if opts.dryRun {
+		fmt.Println(string(manifests))
+		return nil
+	}
+
+	return opts.kubeFactory.Apply(ctx, "", manifests)
+}
+
+func createAddClusterKustomization(ingressUrl, contextName, server, csdpToken string, version *string) *kusttypes.Kustomization {
```

Comment:
```suggestion
func createAddClusterKustomization(ingressUrl, contextName, server, csdpToken string, version string) *kusttypes.Kustomization {
```

Think it's better to dereference in the calling function